molecule/scenario.py

```diff
@@ -240,16 +237,11 @@ def ephemeral_directory(path=None):
     """
     d = os.getenv('MOLECULE_EPHEMERAL_DIRECTORY')
     if not d:
-        # Darwin is the only known platform that returns user-isolated tempdirs
-        # so we do not need to make them unique to each user.
-        if platform.system() == 'Darwin':
-            d = tempfile.gettempdir()
-        else:
-            d = os.path.join(tempfile.gettempdir(), 'mol.' + getpass.getuser())
-    else:
-        d = os.path.abspath(d)
-    if path:
-        d = os.path.abspath(os.path.join(d, path))
-        if not os.path.isdir(d):
-            Path(d).mkdir(mode=0o700, parents=True, exist_ok=True)
+        d = os.getenv("XDG_CACHE_HOME", os.path.expanduser("~/.cache"))
+    d = os.path.abspath(os.path.join(d, path if path else 'molecule'))
```

Comment:
Note: With this new code you can just set ```path='molecule'``` in the function's parameter list instead of setting it to ```None``` in the parameter list and then munging it here.

Reply:
@abadger I am fully aware of that but I decided not to do it because I want to be able to change its behaviour/location without having to the API. Now the api has default to None, which means `auto`. 

Makes sense?

Reply:
Hmmmm....  I think we're getting into a tangent here so I do want to stress, I don't think this discussion blocks this change being merged :-)

If the point of an auto location is to allow you to change it without breaking user expectations at a later time, I think it's reasonable to use None as a sentinel value meaning "auto" but that alone isn't going to be obvious to the user.  I think there's several things that you could do to make that apparent:

1. Document that if path is not given, then an *arbitrary* default value will be chosen.
2. Make the location less convenient to discourage someone from depending on it after perusing the filesystem rather than reading the code's documentation.  I notice that some programs using the ~/.cache directory add a hash or base64 encoded string to the filenames they use.  That would be one possibility here.  If you want the directory to be reusable between runs, using something like this would work: ```"molecule-%s" % uuid.uuid5(uuid.NAMESPACE_URL, 'https://github.com/ansible/molecule').hex[:8]``` (Values are arbitrary as we don't actually need something unique here; just something to discourage users from depending on the name being stable if they come across the name on the filesystem.)